src/frontend/main/CityTables.tsx

```diff
@@ -0,0 +1,222 @@
+import React, { useState, useEffect, useRef, useCallback } from 'react';
+import styled from 'styled-components';
+import handleResponseData from './handleResponseData';
+import { NAVHEIGHT, CITYSELECTHEIGHT } from './constants';
+
+type CityTablesProps = {
+  data: any;
+  selectedCity: string;
+  isEmbed: boolean;
+};
+
+type TableViewWrapperProps = {
+  isEmbed: boolean;
+};
+
+const Table = styled.table`
+  width: 100%;
+  table-layout: fixed;
+  border-collapse: collapse;
+  margin-top: 16px;
+
+  th {
+    font-weight: normal;
+
+    &:nth-child(1) {
+      text-align: left;
+    }
+  }
+
+  th,
+  td {
+    padding: 1px 4px;
+    width: 90px;
+
+    @media (min-width: 450px) {
+      padding-left: 0;
+      width: 140px;
+    }
+  }
+
+  th:nth-child(1),
+  td:nth-child(1) {
+    padding-left: 16px;
+    width: auto;
+  }
+
+  th:nth-child(3),
+  td:nth-child(3) {
+    padding-right: 16px;
+  }
+
+  td {
+    text-align: center;
+  }
+
+  tbody tr:nth-child(1) {
+    th,
+    td {
+      padding-top: 8px;
+    }
+  }
+`;
+
+const TableContainer = styled.div`
+  width: 100%;
+  border-bottom: 1px solid ${props => props.theme.grey};
+  padding: 0 0 24px 0;
+
+  p {
+    margin: 0 0 4px 0;
+  }
+`;
+
+const CityHeadingContainer = styled.div`
+  padding: 0 16px;
+`;
+
+const BoldText = styled.p`
+  font-weight: bold;
+`;
+
+const ItalicText = styled.p`
+  font-style: italic;
+`;
+
+const CityHeading = styled.h2`
+  font-size: 21px;
+  margin: 18px 0 8px;
+`;
+
+const TableHead = styled.thead`
+  background: ${props => props.theme.lightGrey};
+
+  th {
+    padding: 6px 4px;
+    font-style: italic;
+    font-size: 14px;
+  }
+`;
+
+const Loading = styled.p`
+  margin-top: 40px;
+`;
+
+const TableViewWrapper = styled.div<TableViewWrapperProps>`
+  max-width: 600px;
+  margin: 0 auto;
+  padding-bottom: 40px;
+  height: ${props => (props.isEmbed ? `calc(100vh - (${CITYSELECTHEIGHT}px + ${NAVHEIGHT}px))` : 'auto')};
+  overflow: ${props => (props.isEmbed ? 'auto' : 'initial')};
+`;
+
+const CityTables = ({ data, selectedCity, isEmbed }: CityTablesProps) => {
+  const [listItems, setListItems] = useState(data.slice(0, 10));
+  const [isFetching, setIsFetching] = useState(false);
+  const container = useRef<HTMLDivElement>(null);
+
+  const fetchMoreListItems = useCallback(() => {
+    if (isFetching) return;
+    setIsFetching(true);
+
+    setTimeout(() => {
```

Comment:
What is the intention of slowing down the "fetching" process with `setTimeout` here? :thinking:

Reply:
I was wondering about that too, but now checked the reference article again: 

> For demo purposes, I’ve wrapped the setter inside a timeout to simulate two seconds of loading time that you might experience from a real-world application.

🙈🤦‍♀️😅

Reply:
:laughing: